CPS-0011/README.md

```diff
@@ -0,0 +1,64 @@
+---
+CPS: 11
+Title: Universal JSON Encoding for Domain Types
+Status: Open
+Category: Tools
+Authors:
+    - Vladimir Kalnitsky <vladimir@mlabs.city>
+Proposed Solutions: []
+Discussions:
+    - https://github.com/cardano-foundation/cips/pulls/742
+Created: 2024-01-10
+---
+
+## Abstract
+
+dApps require the use of communication protocols that facilitate data transfer between:
+
+- query layer providers and dApps
+- wallets and dApps
+- wallets and query layer providers
+- dApp frontends and backends
+- different dApps
+- Cardano blockchain indexers and their API consumers
+
+Usually these protocols include a specification of Cardano domain types, i.e. ledger block and all of its sub-components.
+
+## Problem
+
+Cardano domain types have canonical CDDL definitions (for every era), but when it comes to use in web apps, where JSON is the universally accepted format, there is no definite standard.
+
+As a result, software solutions are incompatible with each other, and dApp developers are forced to write code for conversions that could in princible be unnecessary, because the semantics of different JSON layouts are often the same. In particular, this problem is very real when offchain libraries need to provide support for different query layer providers (examples: [Lucid](https://lucid.spacebudz.io/docs/getting-started/choose-provider/), [Mesh.js](https://meshjs.dev/providers), [cardano-transaction-lib](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/runtime.md), [Atlas](https://haddock.atlas-app.io/GeniusYield-Providers.html)).
```

Comment:
```suggestion
As a result, software solutions are incompatible with each other, and dApp developers are forced to write code for conversions that could in principle be unnecessary, because the semantics of different JSON layouts are often the same. In particular, this problem is very real when offchain libraries need to provide support for different query layer providers (examples: [Lucid](https://lucid.spacebudz.io/docs/getting-started/choose-provider/), [Mesh.js](https://meshjs.dev/providers), [cardano-transaction-lib](https://github.com/Plutonomicon/cardano-transaction-lib/blob/develop/doc/runtime.md), [Atlas](https://haddock.atlas-app.io/GeniusYield-Providers.html)).
```